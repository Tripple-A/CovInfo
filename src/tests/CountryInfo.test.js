import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import CountryInfo from '../components/countryInfo';


const div = document.createElement('div');
const match = {
  params: {
    name: 'Albania',
  },
};

it('renders without crashing', () => {
  render(
    <MemoryRouter>
      <CountryInfo match={match} />
    </MemoryRouter>, div,
  );
});

it('renders text correctly', () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <CountryInfo match={match} />
    </MemoryRouter>, div,
  );
  expect(getByTestId('total')).toHaveTextContent('Total');
});

it('matches snapshot 1', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <CountryInfo match={match} />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
