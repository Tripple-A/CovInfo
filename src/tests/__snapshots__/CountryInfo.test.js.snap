// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div>
  <h1
    className="h1"
  />
  <div
    className="cases"
  >
    <div
      className="first"
    >
      <h5
        data-testid="total"
      >
        Total Cases:
      </h5>
      <h5>
        <span
          className="red"
        >
          Total Deaths: 
        </span>
      </h5>
      <h5>
        <span
          className="red"
        >
          \`Today's\` Cases:
        </span>
      </h5>
      <h5>
        <span
          className="green"
        >
           Cases recovered: 
        </span>
      </h5>
    </div>
    <div>
      <h5>
        <span
          className="red"
        >
          \`Today's\` Deaths: 
        </span>
      </h5>
      <h5>
        Active Cases:
      </h5>
      <h5>
        <span
          className="green"
        >
          Tests Carried Out: 
        </span>
      </h5>
      <h5>
        <span
          className="red"
        >
          Critical Cases: 
        </span>
      </h5>
    </div>
  </div>
  <div
    className="btn"
  >
    <a
      href="/"
      onClick={[Function]}
    >
       
      <button
        className="btn btn-primary"
        type="button"
      >
        Back
      </button>
       
    </a>
  </div>
</div>
`;
